src/Migration.php

```diff
@@ -146,8 +146,8 @@ public static function of($source, $params = []): self
     public static function register(string $driverType, string $migrator = null)
     {
         // forward to generic Migration::register if called with a descendant class e.g Migration\Mysql::register
-        if (static::class !== __CLASS__) {
-            return call_user_func([__CLASS__, 'register'], $driverType, $migrator ?: static::class);
+        if (static::class !== self::class) {
```

Comment:
It is tricky to mess with this here

Reply:
wdym? it is very clean!

Reply:
I remember it was tricky to set this up so you can call it from child classes, etc.
If test pass should be fine

Reply:
`__CLASS__` was very hard to read, self::class vs static::class is non-forward thus absolutely safe

I do not see any single issue here

Reply:
yes, there was actually an issue, it seems that even `self::` refers to get_class($this)... I may send a RFC for this to PHP... now fixed